ssg/build_remediations.py

```diff
@@ -180,6 +113,43 @@ def expand_env_yaml_from_rule(self):
     def parse_from_file_with_jinja(self, env_yaml, cpe_platforms):
         return parse_from_file_with_jinja(self.file_path, env_yaml)
 
+    def get_inherited_cpe_platform_names(self):
+        inherited_cpe_platform_names = set()
+        if self.associated_rule:
+            # There can be repeated inherited platforms and rule platforms
+            inherited_cpe_platform_names.update(self.associated_rule.inherited_cpe_platform_names)
+        return inherited_cpe_platform_names
+
+    def get_rule_specific_cpe_platform_names(self):
+        rule_specific_cpe_platform_names = set()
+        inherited_cpe_platform_names = self.get_inherited_cpe_platform_names()
+        if self.associated_rule:
+            if self.associated_rule.cpe_platform_names is not None:
+                rule_specific_cpe_platform_names = {
+                    p for p in self.associated_rule.cpe_platform_names
+                    if p not in inherited_cpe_platform_names}
+        return rule_specific_cpe_platform_names
+
+    def get_rule_specific_conditionals(self, language, cpe_platforms):
```

Comment:
Decompose those methods - you retrieve something, and then you perform an operation over the result - you get a remediation that you strip. There is no need to have two almost identical methods - have two thin, and one separate with the common code.